### PR TITLE
test: Port interactive transaction tests to new setup

### PR DESCRIPTION
**Pull request by SevInf**
Close prisma/client-planning#23

### CODE REVIEW

File: packages/client/tests/functional/interactive-transactions/__snapshots__/tests.ts.snap [hunk 1]
```diff
@@ -0,0 +1,121 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`interactive-transactions (mongodb) () batching rollback 1`] = `
```
Comment (SevInf) [inline packages/client/tests/functional/interactive-transactions/__snapshots__/tests.ts.snap:3]: All these snapshots have different output per provider, so I had to convert inline snapshots into external ones.

File: packages/client/tests/functional/interactive-transactions/tests.ts [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+// @ts-ignore this is just for type checks
+declare let PrismaClient: typeof import('@prisma/client').PrismaClient
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
 const testIf = (condition: boolean) => (condition ? test : test.skip)
 
-let PrismaClient, prisma
+testMatrix.setupTestSuite(({ provider }) => {
+  beforeEach(async () => {
+    await prisma.user.deleteMany()
```
Comment (danstarns) [inline packages/client/tests/functional/interactive-transactions/tests.ts:15]: After each test, we drop the DB https://github.com/prisma/prisma/blob/main/packages/client/tests/functional/_utils/setupTestSuiteMatrix.ts#L62, and foreach test we setup a new db, does this need to be done?

File: packages/client/tests/functional/_utils/setupFilesAfterEnv.ts [hunk 3]
```diff
@@ -3,5 +3,7 @@ process.env.DATABASE_URI_sqlite = 'file:dev.db'
 process.env.DATABASE_URI_mongodb = process.env.TEST_MONGO_URI
 process.env.DATABASE_URI_postgresql = process.env.TEST_POSTGRES_URI
 process.env.DATABASE_URI_mysql = process.env.TEST_MYSQL_URI
+process.env.DATABASE_URI_cockroachdb = process.env.TEST_COCKROACH_URI
```
Comment (danstarns) [inline packages/client/tests/functional/_utils/setupFilesAfterEnv.ts:6]: Nice work on adding these, I noticed however you arent testing either, can we add coverage for both of these providers in this test?

File: packages/client/tests/functional/interactive-transactions/tests.ts (hunk 2 above)
Comment (danstarns) [reply packages/client/tests/functional/interactive-transactions/tests.ts:15]: Would using unique values help, such as using something like faker?

Comment (SevInf) [reply packages/client/tests/functional/interactive-transactions/tests.ts:15]: I think it might, let me try that

Comment (danstarns) [reply packages/client/tests/functional/interactive-transactions/tests.ts:15]: Thanks for checking!

**Comment by SevInf** [comment]
Going to draft it until all CI issues are fixed

**Comment by SevInf** [comment]
Needs prisma/client-planning#65 to proceed

File: packages/client/tests/functional/interactive-transactions/prisma/_schema.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    provider = "${provider}"
+    url      = env("DATABASE_URI_${provider}")
+  }
+  
+
+  // / User model comment
+  model User {
+    id    ${id}
+    email String  @unique
+    // / name comment
```
Comment (millsp) [inline packages/client/tests/functional/interactive-transactions/prisma/_schema.ts:21]: I think we can clean these comments every time you see them. These don't work in this form, anyways.

**Review by millsp — APPROVED** [review]

**Pull request merged**